plugins/trip.hpp

```diff
@@ -0,0 +1,332 @@
+        auto prefix = 0;
+        for (size_t j = 0; j < scc.get_number_of_components(); ++j){
+            range_insertion.push_back(prefix);
+            range.push_back(prefix);
+            prefix += scc.get_component_size(j);
+        }
+
+        for (size_t i = 0; i < number_of_locations; ++i) {
+            components[range_insertion[scc.get_component_id(i)]] = i;
+            ++range_insertion[scc.get_component_id(i)];
+        }
+
+        return SCC_Component(components, range);
+    }
+
+    void SetLocPermutationOutput(const std::vector<NodeID> & permutation,
+                                 osrm::json::Object & json_result){
+        osrm::json::Array json_permutation;
+        json_permutation.values.insert(std::end(json_permutation.values),
+                                       std::begin(permutation),
+                                       std::end(permutation));
+        json_result.values["permutation"] = json_permutation;
+    }
+
+    void ComputeRoute(const PhantomNodeArray & phantom_node_vector,
+                      const RouteParameters & route_parameters,
+                      const std::vector<NodeID> & trip,
+                      InternalRouteResult & min_route) {
+        // given he final trip, compute total distance and return the route and location permutation
+        PhantomNodes viapoint;
+        for (auto it = std::begin(trip); it != std::end(trip); ++it) {
+            const auto from_node = *it;
+            // if from_node is the last node, compute the route from the last to the first location
+            const auto to_node = std::next(it) != std::end(trip) ? *std::next(it) : *std::begin(trip);
+
+            viapoint = PhantomNodes{phantom_node_vector[from_node][0], phantom_node_vector[to_node][0]};
+            min_route.segment_end_coordinates.emplace_back(viapoint);
+        }
+
+        search_engine_ptr->shortest_path(min_route.segment_end_coordinates, route_parameters.uturns, min_route);
+
+        BOOST_ASSERT_MSG(min_route.shortest_path_length < INVALID_EDGE_WEIGHT,
+                         "unroutable route");
+    }
+
+    int HandleRequest(const RouteParameters &route_parameters,
+                      osrm::json::Object &json_result) override final
+    {
+        // check if all inputs are coordinates
+        if (!check_all_coordinates(route_parameters.coordinates)) {
+            return 400;
+        }
+
+        // get phantom nodes
+        PhantomNodeArray phantom_node_vector(route_parameters.coordinates.size());
+        GetPhantomNodes(route_parameters, phantom_node_vector);
+        auto number_of_locations = phantom_node_vector.size();
+
+        // compute the distance table of all phantom nodes
+        const auto result_table = DistTableWrapper<EdgeWeight>(*search_engine_ptr->distance_table(phantom_node_vector),
+                                                               number_of_locations);
+        if (result_table.size() == 0){
+            return 400;
+        }
```

Comment:
You can already check this before wrapping the distance table into your `DistTableWrapper`.